dangerzone/isolation_provider/container.py

```diff
@@ -228,20 +228,28 @@ def kill_container(self, name: str) -> None:
         container_runtime = self.get_runtime()
         cmd = [container_runtime, "kill", name]
         try:
+            # We do not check the exit code of the process here, since the container may
+            # have stopped right before invoking this command. In that case, the
+            # command's output will contain some error messages, so we capture them in
+            # order to silence them.
             subprocess.run(
-                cmd,
-                capture_output=True,
-                check=True,
-                startupinfo=get_subprocess_startupinfo(),
+                cmd, capture_output=True, startupinfo=get_subprocess_startupinfo()
             )
-        except subprocess.CalledProcessError as e:
-            log.warning(f"Failed to kill container {name}: {str(e)}")
-            log.warning(f"Output from the kill command: {e.output}")
         except Exception as e:
             log.exception(
-                f"Unexpected error occurred while killing container {name}: {str(e)}"
+                f"Unexpected error occurred while killing container '{name}': {str(e)}"
             )
 
+        # Sanity check to see if the container has indeed exited. It has a unique ID, so
+        # it's safe to check for it's name in the command's output.
+        all_containers = subprocess.run(
+            [container_runtime, "ps", "-a"],
+            capture_output=True,
+            startupinfo=get_subprocess_startupinfo(),
+        )
+        if name in all_containers.stdout.decode():
+            log.warning(f"Container '{name}' did not stop gracefully")
```

Comment:
Oh, sorry for the confusion. I'll clarify a bit on what I'm searching for here. If the description makes sense, I'll add it as a comment.

> I am a bit confused by what we're trying to achieve here witht the warning statement. Are we try to find if the container was indeed killed or if didn't show up because it died before it was killed?

We're trying to see if the container no longer exists. Either because we successfully killed the container, or because the container exited on its own. We operate under the assumption that after a `podman kill` / `docker kill` invocation, this will likely be the case, else the container runtime (Docker/Podman) has experienced a problem, and we should report it.

Two implementation details here worthy of mention:
1. We use `--rm` when starting our containers, meaning that they should be removed every time we kill them.
2. We could use a `--filter` argument, but I figured that the container ID should be unique, so it's not worth the hassle.

Does this answer your question?

Reply:
Cool, I'll let you know once I have something.

Reply:
I have a fix for that in 0c74063.